### PR TITLE
Visitor can see categorized articles

### PR DESCRIPTION
**Pull request by vick3d**
#### Description
As a Visitor
In order to find relevant articles
I would like for the articles to be categorized

Changes proposed in this pull request:

* Add category to article
* Sort articles by category in index page

#### What I have learned working on this feature:
Association and iteration between models

#### Screenshots:
image

### CODE REVIEW

File: features/step_definitions/assertion_steps.rb [hunk 1]
```diff
@@ -22,4 +22,11 @@
 	within("#article_#{@article.id}") do
 		expect(page).not_to have_content expected_content
 	end
+end
+
+Then("I should see {string} in {string}") do |expected_content, name|
+	@category = Category.find_by_name(name)
```
Comment (oliverochman) [inline features/step_definitions/assertion_steps.rb]: do you need to use an instance variable here?

File: features/step_definitions/basic_steps.rb [hunk 2]
```diff
@@ -1,6 +1,14 @@
 Given("the following articles exist") do |table|
-	table.hashes.each do |article|
-		create(:article, article)
+  table.hashes.each do |article|
+    category = Category.find_by(name: article[:category])
```
Comment (oliverochman) [inline features/step_definitions/basic_steps.rb]: Change this to `Category.find_or_create_by(name)`. If you do this you can remove the step about creating the categories

File: spec/models/category_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    it { is_expected.to have_db_column :name }
+  end
+
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :name }
+    it { is_expected.to validate_uniqueness_of :name }
+		it { should_not allow_value("weather").for(:name) }
+		it { should allow_value("sports").for(:name) }
+	end
+
```
Comment (oliverochman) [inline spec/models/category_spec.rb]: You need to test the association between the articles and category in both of the unit specs.

Comment (LauraRe) [reply spec/models/category_spec.rb]: done

File: features/step_definitions/basic_steps.rb (hunk 2 above)
Comment (LauraRe) [reply features/step_definitions/basic_steps.rb]: done

File: features/step_definitions/assertion_steps.rb (hunk 1 above)
Comment (LauraRe) [reply features/step_definitions/assertion_steps.rb]: done

File: spec/models/category_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	describe 'DB table' do
+    it { is_expected.to have_db_column :name }
+  end
+
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :name }
+    it { is_expected.to validate_uniqueness_of :name }
+		it { should_not allow_value("weather").for(:name) }
+		it { should allow_value("sports").for(:name) }
+		it { should have_many(:articles) }
```
Comment (oliverochman) [inline spec/models/category_spec.rb]: This test should be in an 'Associations' describe block

Comment (LauraRe) [reply spec/models/category_spec.rb]: done

**Pull request merged**